### PR TITLE
Extract certificate add token metadata from remote member

### LINKED ISSUE #13732: Using a client join token only works when talking to the cluster member that generated it
In a LXD cluster, if one member does `lxc config trust add` to generate a token, then this can only be used by the client running `lxc remote add` if the remote server it specifies is the specific cluster member that issued the token.

But this should not be the case, the LXD cluster member receiving the client join request should lookup the join token details from the cluster member that generated the token via its local operation data.

```
root@micro01:~# lxc config trust add --name clientfoo
If this is your first time running LXD on this machine, you should also run: lxd init
To start your first container, try: lxc launch ubuntu:22.04
Or for a virtual machine: lxc launch ubuntu:22.04 --vm

Client clientfoo add token:
eyJjbGllbnRfbmFtZSI6InRlbC1taWNyb2Nsb3VkLnRhaXBlaS5pbnRlcm5hbCIsImZpbmdlcnByaW50IjoiYTBlMzc1MDk0NzkyYTNjZDM1MzkzZTNmYTc0MTA4ZGQyODE1ODg0Yjc4MTRmY2U3Y2RiNmZlY2U2ODNmYTViNyIsImFkZHJlc3NlcyI6WyIxMC4xOTMuOTYuMTU6ODQ0MyIsIltmZDQyOmY5ZjE6NmUwMjoxMjk5OjIxNjozZWZmOmZlMjQ6NjA4ZF06ODQ0MyJdLCJzZWNyZXQiOiJjNDM4ZTBhYjUwM2JmM2E1OWE1MTc2NjA1YmZhZjFhZDZiYWUxOWJlMzQ1ZGMxM2MzODUxMmZlZWY2MTExZjczIiwiZXhwaXJlc19hdCI6IjAwMDEtMDEtMDFUMDA6MDA6MDBaIn0=
```

```
root@c1:~# lxc remote add mylxdcluster micro02.lxd 
Admin password (or token) for mylxdcluster: 
Error: Bad certificate add operation data
```

### PR DESCRIPTION
**Pull request by markylaing**
When a certificate add operation is returned by another member, the operation metadata is unmarshalled into a map[string]any. Since there is no hint that the value of `Metadata["request"]` should be an `api.CertificatesPost`, the contents will be another `map[string]any` following json unmarshalling defaults.

So that we do not encounter issues with diverging field names if the `api.CertificatesPost` type changes, I have opted to marshal the `any` that is in `Metadata["request"]`, and subsequently unmarshal it into the correct type.

Closes #13732

### CODE REVIEW

**Comment by markylaing** [comment]
> Thanks!
> 
> Could we do this wrangling inside `certificateTokenValid` and have it return a `*api.CertificatesPost` to avoid the caller having to care anything about how the operation is decoded.

Done.

I've tested this locally by creating a two node cluster, getting a join token from one member, then using the URL of the other member when adding the remote. Should I also add a test for this somewhere in the clustering suite? Not sure how easy it'll be.

**Comment by tomponline** [comment]
> Should I also add a test for this somewhere in the clustering suite

That would be great if not too involved

File: lxd/certificates.go [hunk 1]
```diff
@@ -449,10 +479,8 @@ func certificatesPost(d *Daemon, r *http.Request) response.Response {
 				return response.Forbidden(fmt.Errorf("No matching certificate add operation found"))
 			}
 
-			tokenReq, ok := joinOp.Metadata["request"].(api.CertificatesPost)
-			if !ok {
-				return response.InternalError(fmt.Errorf("Bad certificate add operation data"))
-			}
+			// We have already validated that the request metadata is present and ensured it is this type in certificateTokenValid.
```
Comment (tomponline) [inline lxd/certificates.go:482]: Can we just return a api.CertificatesPost from certificateTokenValid rather than an operation?

Comment (markylaing) [reply lxd/certificates.go:482]: Ah yeah sorry I missed that part

Comment (markylaing) [reply lxd/certificates.go]: Done.

**Comment by markylaing** [comment]
> > Should I also add a test for this somewhere in the clustering suite
> 
> That would be great if not too involved

Ok I'll give it a go tomorrow morning.

**Review by tomponline — APPROVED** [review]
Thanks!

**Comment by markylaing** [comment]
Tests added in https://github.com/canonical/lxd/pull/13740

**Pull request merged**